client/client.go

```diff
@@ -200,6 +206,17 @@ func (e ConnectionError) Unwrap() error {
 	return e.Err
 }
 
+type InternalClientError struct{ Err error }
+
+func (e InternalClientError) Error() string {
```

Comment:
I was thinking that maybe we should use pointer receivers, but I see that there's a couple of other errors here that already use values in Error(), so it's probably not worth it.